### PR TITLE
chore: fix linter configurations in `frontend`

### PR DESCRIPTION
**Pull request by stephenliang**
Some linter configurations were missing in the packages resulting in the storybook app and some other packages not being formatted by prettier. This PR adds those missing configurations.

Additionally, a new script `release:dryrun` was added that will execute lint, prettier, test, and build to allow for local one step pre-PR quality testing.

### CODE REVIEW

**Review by kelbyhawn — APPROVED** [review]
This lgtm, but someone else should look too!

File: frontend/apps/design-system-storybook/package.json [hunk 1]
```diff
@@ -31,22 +50,15 @@
     "@storybook/test-runner": "^0.19.1",
     "axe-playwright": "^2.0.3",
     "concurrently": "^9.1.0",
+    "eslint": "^9.16.0",
+    "globals": "^15.13.0",
     "http-server": "^14.1.1",
+    "prettier": "3.3.3",
```
Comment (drizco) [inline frontend/apps/design-system-storybook/package.json:56]: intentional to pin to 3.3.3?

**Review by drizco — APPROVED** [review]
one question but lgtm!

File: frontend/apps/design-system-storybook/package.json (hunk 1 above)
Comment (stephenliang) [reply frontend/apps/design-system-storybook/package.json:56]: This seems to be the [recommendation from prettier](https://prettier.io/docs/en/install.html):

> Install an exact version of Prettier locally in your project. This makes sure that everyone in the project gets the exact same version of Prettier. Even a patch release of Prettier can result in slightly different formatting, so you wouldn’t want different team members using different versions and formatting each other’s changes back and forth.

**Pull request merged**